From: Christoph Krapp <achterin@googlemail.com>
Date: Thu, 13 Dec 2018 14:49:47 +0100
Subject: ramips: mt7620: fix mt76 driver selection

40692f0fb5 ramips: mt7620: select only the matching mt76 driver
f381cbcb5f ramips: fix Netgear EX3700/EX3800 wireless driver selection
7a32a73e52 ramips: mt7620: drop mt76 driver for boards without mt76 wireless

Signed-off-by: Christoph Krapp <achterin@googlemail.com>

diff --git a/target/linux/ramips/image/mt7620.mk b/target/linux/ramips/image/mt7620.mk
index f627627192bf4899392bd7079e809f05e6e9ce5a..c34eb4d7dbfb218dac683c6bad8dbf1265aca4a3 100644
--- a/target/linux/ramips/image/mt7620.mk
+++ b/target/linux/ramips/image/mt7620.mk
@@ -32,7 +32,7 @@ define Device/alfa-network_ac1200rm
   DTS := AC1200RM
   IMAGE_SIZE := 16064k
   DEVICE_TITLE := ALFA Network AC1200RM
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci uboot-envtools
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci uboot-envtools
 endef
 TARGET_DEVICES += alfa-network_ac1200rm
 
@@ -54,6 +54,7 @@ define Device/ArcherC20i
   TPLINK_HWREV := 58
   IMAGES += factory.bin
   DEVICE_TITLE := TP-Link ArcherC20i
+  DEVICE_PACKAGES := kmod-mt76x0e
 endef
 TARGET_DEVICES += ArcherC20i
 
@@ -68,6 +69,7 @@ define Device/ArcherC50v1
   IMAGE/factory-us.bin := tplink-v2-image -e -w 0
   IMAGE/factory-eu.bin := tplink-v2-image -e -w 2
   DEVICE_TITLE := TP-Link ArcherC50v1
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += ArcherC50v1
 
@@ -78,7 +80,7 @@ define Device/ArcherMR200
   TPLINK_FLASHLAYOUT := 8MLmtk
   TPLINK_HWID := 0xd7500001
   TPLINK_HWREV := 0x4a
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-net kmod-usb-net-rndis kmod-usb-serial kmod-usb-serial-option adb-enablemodem
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-net kmod-usb-net-rndis kmod-usb-serial kmod-usb-serial-option adb-enablemodem
   DEVICE_TITLE := TP-Link ArcherMR200
 endef
 TARGET_DEVICES += ArcherMR200
@@ -117,7 +119,7 @@ define Device/d240
   DTS := D240
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Sanlinking Technologies D240
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76-core kmod-mt76x2 kmod-sdhci-mt7620
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620
 endef
 TARGET_DEVICES += d240
 
@@ -143,7 +145,7 @@ define Device/dch-m225
 	seama-seal -m "signature=wapn22_dlink.2013gui_dap1320b" | \
 	check-size $$$$(IMAGE_SIZE)
   DEVICE_TITLE := D-Link DCH-M225
-  DEVICE_PACKAGES := kmod-mt76 kmod-sound-core kmod-sound-mt7620 kmod-i2c-ralink
+  DEVICE_PACKAGES := kmod-sound-core kmod-sound-mt7620 kmod-i2c-ralink
 endef
 TARGET_DEVICES += dch-m225
 
@@ -211,7 +213,6 @@ define Device/ex2700
   KERNEL := $(KERNEL_DTB) | uImage lzma | pad-offset 64k 64 | append-uImage-fakehdr filesystem
   IMAGE/factory.bin := $$(sysupgrade_bin) | check-size $$$$(IMAGE_SIZE) | \
 	netgear-dni
-  DEVICE_PACKAGES := -kmod-mt76 -kmod-mt7603 -kmod-mt76x2 -kmod-mt76-core
   DEVICE_TITLE := Netgear EX2700
 endef
 TARGET_DEVICES += ex2700
@@ -223,7 +224,7 @@ define Device/ex3700-ex3800
   IMAGE_SIZE := 7744k
   IMAGES += factory.chk
   IMAGE/factory.chk := $$(sysupgrade_bin) | check-size $$$$(IMAGE_SIZE) | netgear-chk
-  DEVICE_PACKAGES := -kmod-mt76 kmod-mt76x2
+  DEVICE_PACKAGES := kmod-mt76x2
   DEVICE_TITLE := Netgear EX3700/EX3800
   SUPPORTED_DEVICES := ex3700
 endef
@@ -233,7 +234,7 @@ define Device/gl-mt300a
   DTS := GL-MT300A
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := GL-Inet GL-MT300A
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci
 endef
 TARGET_DEVICES += gl-mt300a
 
@@ -248,7 +249,7 @@ define Device/gl-mt300n
   DTS := GL-MT300N
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := GL-Inet GL-MT300N
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci
 endef
 TARGET_DEVICES += gl-mt300n
 
@@ -256,7 +257,7 @@ define Device/gl-mt750
   DTS := GL-MT750
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := GL-Inet GL-MT750
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci
 endef
 TARGET_DEVICES += gl-mt750
 
@@ -272,7 +273,7 @@ define Device/hc5761
   DTS := HC5761
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := HiWiFi HC5761
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 kmod-usb-ledtrig-usbport
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 kmod-usb-ledtrig-usbport
 endef
 TARGET_DEVICES += hc5761
 
@@ -280,7 +281,7 @@ define Device/hc5861
   DTS := HC5861
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := HiWiFi HC5861
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 kmod-usb-ledtrig-usbport
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 kmod-usb-ledtrig-usbport
 endef
 TARGET_DEVICES += hc5861
 
@@ -329,7 +330,7 @@ define Device/miwifi-mini
   DTS := MIWIFI-MINI
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Xiaomi MiWiFi Mini
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci
 endef
 TARGET_DEVICES += miwifi-mini
 
@@ -377,7 +378,7 @@ TARGET_DEVICES += mt7620a_v22sg
 define Device/mzk-750dhp
   DTS := MZK-750DHP
   DEVICE_TITLE := Planex MZK-750DHP
-  DEVICE_PACKAGES := kmod-mt76
+  DEVICE_PACKAGES := kmod-mt76x0e
 endef
 TARGET_DEVICES += mzk-750dhp
 
@@ -390,7 +391,7 @@ TARGET_DEVICES += mzk-ex300np
 define Device/mzk-ex750np
   DTS := MZK-EX750NP
   DEVICE_TITLE := Planex MZK-EX750NP
-  DEVICE_PACKAGES := kmod-mt76
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += mzk-ex750np
 
@@ -411,7 +412,7 @@ TARGET_DEVICES += oy-0001
 define Device/psg1208
   DTS := PSG1208
   DEVICE_TITLE := Phicomm PSG1208
-  DEVICE_PACKAGES := kmod-mt76
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += psg1208
 
@@ -461,14 +462,14 @@ define Device/rt-ac51u
   DTS := RT-AC51U
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Asus RT-AC51U
-  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ehci kmod-usb-ledtrig-usbport
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb-core kmod-usb2 kmod-usb-ehci kmod-usb-ledtrig-usbport
 endef
 TARGET_DEVICES += rt-ac51u
 
 define Device/tiny-ac
   DTS := TINY-AC
   DEVICE_TITLE := Dovado Tiny AC
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb2 kmod-usb-ohci
 endef
 TARGET_DEVICES += tiny-ac
 
@@ -481,7 +482,7 @@ define Device/tplink_c20-v1
   TPLINK_HWREVADD := 0x1
   IMAGES += factory.bin
   DEVICE_TITLE := TP-Link ArcherC20 v1
-  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  DEVICE_PACKAGES := kmod-mt76x0e kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
 endef
 TARGET_DEVICES += tplink_c20-v1
 
@@ -497,7 +498,7 @@ define Device/ravpower_wd03
   DTS := WD03
   IMAGE_SIZE := $(ralink_default_fw_size_8M)
   DEVICE_TITLE := Ravpower WD03
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76 kmod-usb-ehci
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-usb-ehci
 endef
 TARGET_DEVICES += ravpower_wd03
 
@@ -505,6 +506,7 @@ define Device/whr-1166d
   DTS := WHR-1166D
   IMAGE_SIZE := 15040k
   DEVICE_TITLE := Buffalo WHR-1166D
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += whr-1166d
 
@@ -586,6 +588,7 @@ define Device/y1
   DTS := Y1
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Lenovo Y1
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += y1
 
@@ -593,6 +596,7 @@ define Device/y1s
   DTS := Y1S
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Lenovo Y1S
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += y1s
 
@@ -609,14 +613,14 @@ define Device/we1026-5g-16m
   IMAGE_SIZE := 16777216
   SUPPORTED_DEVICES += we1026-5g-16m
   DEVICE_TITLE := Zbtlink ZBT-WE1026-5G (16M)
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76 kmod-sdhci-mt7620
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620
 endef
 TARGET_DEVICES += we1026-5g-16m
 
 define Device/zbt-ape522ii
   DTS := ZBT-APE522II
   DEVICE_TITLE := Zbtlink ZBT-APE522II
-  DEVICE_PACKAGES := kmod-mt76
+  DEVICE_PACKAGES := kmod-mt76x2
 endef
 TARGET_DEVICES += zbt-ape522ii
 
@@ -644,7 +648,7 @@ define Device/zbt-we826-16M
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   SUPPORTED_DEVICES += zbt-we826
   DEVICE_TITLE := Zbtlink ZBT-WE826 (16M)
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76 kmod-sdhci-mt7620 
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620
 endef
 TARGET_DEVICES += zbt-we826-16M
 
@@ -652,7 +656,7 @@ define Device/zbt-we826-32M
   DTS := ZBT-WE826-32M
   IMAGE_SIZE := $(ralink_default_fw_size_32M)
   DEVICE_TITLE := Zbtlink ZBT-WE826 (32M)
-  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-mt76 kmod-sdhci-mt7620
+  DEVICE_PACKAGES := kmod-mt76x2 kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620
 endef
 TARGET_DEVICES += zbt-we826-32M
 
diff --git a/target/linux/ramips/mt7620/target.mk b/target/linux/ramips/mt7620/target.mk
index c4fa2ade75f61ff6395aad9466ad35949c6fcd16..07ed7788fff1fb23a7f52715d3cbdcd3701d5343 100644
--- a/target/linux/ramips/mt7620/target.mk
+++ b/target/linux/ramips/mt7620/target.mk
@@ -7,9 +7,8 @@ BOARDNAME:=MT7620 based boards
 FEATURES+=usb
 CPU_TYPE:=24kc
 
-DEFAULT_PACKAGES += kmod-rt2800-pci kmod-rt2800-soc kmod-mt76 wpad-mini
+DEFAULT_PACKAGES += kmod-rt2800-pci kmod-rt2800-soc wpad-mini
 
 define Target/Description
 	Build firmware images for Ralink MT7620 based boards.
 endef
-
